src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -147,9 +147,13 @@ static void PurgeReadClientContainers(
                     container.readClientPtr = nullptr;
                 }
                 if (container.pathParams) {
-                    Platform::Delete(container.pathParams);
+                    Platform::MemoryFree(container.pathParams);
                     container.pathParams = nullptr;
                 }
+                if (container.eventPathParams) {
+                    Platform::MemoryFree(container.eventPathParams);
```

Comment:
Again, this should be static_asserting that EventPathParams has a trivial destructor.
```suggestion
                    static_assert(std::is_trivially_destructible<EventPathParams>::value, "EventPathParams destructors won't get run");
                    Platform::MemoryFree(container.eventPathParams);
```